framework/validators/UniqueValidator.php

```diff
@@ -234,7 +244,20 @@ private function prepareConditions($targetAttribute, $model, $attribute)
             $conditions = [$targetAttribute => $model->$attribute];
         }
 
-        return $conditions;
+        if (!$model instanceof ActiveRecord) {
+            return $conditions;
+        }
+
+        // Add table prefix for column.
+        $targetClass = $this->getTargetClass($model);
+        $tableName = $targetClass::tableName();
+        $conditionsWithTableName = [];
```

Comment:
IMO `$conditionsWithTableName variable` can be named better and probably shorter.